### PR TITLE
KIN-375 Add slugs to categorie, categorytags and producttags tables

### PR DESCRIPTION
**Pull request by Mauriciocr207**
[KIN-375](https://loopcrack.atlassian.net/browse/KIN-375) Medium [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

It is used url slugs for product filtering on the product catalog page.

### Solution

Add the slug column for storing the slugs values for the entities that are going to be filtered through the product catalog page

### Requirements checklist

- [x] add a column for storing its slug value of the following entities:
 - Categories
 - Category tags
 - Product Tags
- [x] Update the respective migrations, seeders and models to create the respective slug value of its generated data

### Testing

- [x] Here remember make 
```php 
composer install
php spark migrate:refresh // then acept: yes [y]
php spark db:seed MainTestSeeder
``` 
Then you can go to `admin/categoria` and check all crud to verify the operation make on pr [KIN-336](https://github.com/loopcrack-org/Kinub/pull/160) works well. 
- [x] Once you have been created or update a category, go to database and verify that the slugs columns are filled correctly with the crud and seeders.

## Resources

- [Library use to generate slugs](https://github.com/ausi/slug-generator)

## Demo

[KIN-375]: https://loopcrack.atlassian.net/browse/KIN-375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[KIN-336]: https://loopcrack.atlassian.net/browse/KIN-336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CarlosJCD — APPROVED** [review]

**Review by FernandoJoachin — APPROVED** [review]

File: app/Models/CategoryModel.php [hunk 1]
```diff
@@ -64,16 +68,19 @@ public function updateCategory(string $categoryId, array $categoryData)
     {
         try {
             $this->db->transException(true)->transStart();
+            $slugGenerator                = new SlugGenerator();
+            $categoryData['categorySlug'] = $slugGenerator->generate($categoryData['categoryName']);
```
Comment (CarlosJCD) [inline app/Models/CategoryModel.php:72]: why is it necessary to always generate here the slug? is it necessary when, for example, the category name does not change?

**Review by CarlosJCD — APPROVED** [review]
Please check out the comment i left you.

CC: @CheloChidoone

File: app/Models/CategoryModel.php (hunk 1 above)
Comment (Mauriciocr207) [reply app/Models/CategoryModel.php:72]: We need to make a consult to know if a name is changed, and if the name doesn´t change, the slug generated is equal to the  already generated slug. Then, when we update the register, it is the same.

**Review by CarlosMay7 — APPROVED** [review]

File: app/Database/Seeds/ProductTagSeeder.php [hunk 2]
```diff
@@ -11,19 +12,19 @@ public function run()
...
 
         $productTags = [];
 
-        for ($i = 1; $i < 7; $i++) {
-            $productTagsNames = $faker->randomElements($productTagsNames, 2);
-            $productTags[]    = [
-                'ptName'      => $productTagsNames[0],
-                'ptProductId' => $i,
-            ];
+        for ($j = 0; $j < 20; $j++) {
```
Comment (LUGAMAFE) [inline app/Database/Seeds/ProductTagSeeder.php:22]: why 20? can we remove magic numbers please 🙏

File: app/Database/Seeds/ProductTagSeeder.php [hunk 3]
```diff
@@ -11,19 +12,19 @@ public function run()
...
+            $ptName        = $faker->randomElement($productTagsNames);
             $productTags[] = [
-                'ptName'      => $productTagsNames[1],
-                'ptProductId' => $i,
+                'ptName'      => $ptName,
+                'ptSlug'      => $slugGenerator->generate($ptName),
+                'ptProductId' => $faker->numberBetween(1, 20),
```
Comment (LUGAMAFE) [inline app/Database/Seeds/ProductTagSeeder.php:27]: remove magic numbers please

File: app/Database/Seeds/ProductSeeder.php [hunk 4]
```diff
@@ -13,7 +13,7 @@ public function run()
 
         $products = [];
 
-        for ($i = 0; $i < 6; $i++) {
+        for ($i = 0; $i < 20; $i++) {
```
Comment (LUGAMAFE) [inline app/Database/Seeds/ProductSeeder.php:16]: remove magic numbers please

File: app/Database/Seeds/ProductSeeder.php [hunk 5]
```diff
@@ -25,7 +25,7 @@ public function run()
                     'Potencia'              => $faker->randomNumber('3', true) . ' MW',
                     'Comunicación'          => $faker->randomElements(['5G Ipv6', '3G', 'WPAN', 'WPA2']),
                 ]),
-                'productCategoryId' => $faker->numberBetween(1, 2),
+                'productCategoryId' => $faker->numberBetween(1, 6),
```
Comment (LUGAMAFE) [inline app/Database/Seeds/ProductSeeder.php:28]: remove magic numbers please

File: app/Database/Seeds/CategoryTagSeeder.php [hunk 6]
```diff
@@ -2,31 +2,31 @@
...
-        ];
+        $categories = $this->db->query('SELECT * FROM categories')->getResultArray();
 
-        $this->db->table('category_tags')->insertBatch($categoriesTags);
+        $categoryTags  = [];
+        $faker         = Factory::create();
+        $slugGenerator = new SlugGenerator();
+
+        foreach ($categories as $category) {
+            for ($i = 1; $i < 3; $i++) {
```
Comment (LUGAMAFE) [inline app/Database/Seeds/CategoryTagSeeder.php:20]: remove magic numbers please

File: app/Database/Seeds/CategoryTagSeeder.php [hunk 7]
```diff
@@ -2,31 +2,31 @@
...
+                $categoryTagName = $faker->words(3, true);
```
Comment (LUGAMAFE) [inline app/Database/Seeds/CategoryTagSeeder.php:21]: remove magic numbers please

**Review by LUGAMAFE — APPROVED** [review]
Nice just verify the magic numbers please

**Review by Cristianpan — APPROVED** [review]

**Pull request merged**